transport/http/url.go

```diff
@@ -0,0 +1,40 @@
+package http
+
+func JoinPath(a, b string) string {
+	if len(b) == 0 {
+		return a
+	}
+
+	if len(a) == 0 {
+		a = "/"
+	}
+
+	if b[0] == '/' {
+		b = b[1:]
+	}
+
+	if a[len(a)-1] != '/' {
+		return a + "/" + b
+	}
```

Comment:
We can make these `if` checks consistent with the one's in JoinRawQuery. Or vice-versa.